### DISCUSSION TITLE
課題10についての質問

### DISCUSSION THREAD

**Discussion by TakedaAyano** [nsc-kiwamu/traning-takeda#33] [Q&A]
課題10内の「問題6」についてなのですが、動物の名前だけでなく動物の種類も出力されてしまうのはなぜなのでしょうか？
「問題7」では、名前しか出力されなかったので、原因がわかりませんでした。

**Comment by nsc-kiwamu** [comment]
selectAnyTypeの中でtype(catやdog)とname(ライオンや犬)を配列に入れてしまっているからです。
また問題7はselectAnyTypeの戻り値で受け取っているdogs2を使っていないので
一見うまくいっているように見えてしまっています。
